### PR TITLE
Added the ability to perform automated spell checking

### PR DESCRIPTION
**Pull request by szmazurek**
Fixes #ISSUE_NUMBER

## Proposed Changes
- added CI for spellchecker 

## Checklist
- [ ] [`CONTRIBUTING`](https://github.com/mlcommons/GaNDLF/blob/master/CONTRIBUTING.md) guide has been followed.
- [ ] PR is based on the [current GaNDLF master ](https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/keeping-your-local-repository-in-sync-with-github/syncing-your-branch-in-github-desktop?platform=windows).
- [ ] Non-breaking change (does **not** break existing functionality): provide **as many** details as possible for _any_ breaking change.
- [ ] Function/class source code documentation added/updated (ensure `typing` is used to provide type hints, including and not limited to using `Optional` if a variable has a pre-defined value).
- [ ] Code has been [blacked](https://github.com/psf/black#usage) for style consistency and linting.
- [ ] If applicable, version information [has been updated in GANDLF/version.py](https://github.com/mlcommons/GaNDLF/blob/master/GANDLF/version.py).
- [ ] If adding a git submodule, add to list of exceptions for black styling in [pyproject.toml](https://github.com/mlcommons/GaNDLF/blob/master/pyproject.toml) file.
- [ ] [Usage documentation](https://github.com/mlcommons/GaNDLF/blob/master/docs) has been updated, if appropriate.
- [ ] Tests added or modified to [cover the changes](https://app.codecov.io/gh/mlcommons/GaNDLF); if coverage is reduced, please give explanation.
- [ ] If customized dependency installation is required (i.e., a separate `pip install` step is needed for PR to be functional), please ensure it is reflected in all the files that control the CI, namely: [python-test.yml](https://github.com/mlcommons/GaNDLF/blob/master/.github/workflows/python-test.yml), and all docker files [[1](https://github.com/mlcommons/GaNDLF/blob/master/Dockerfile-CPU),[2](https://github.com/mlcommons/GaNDLF/blob/devcontainer_build_fix/Dockerfile-CUDA11.6),[3](https://github.com/mlcommons/GaNDLF/blob/master/Dockerfile-ROCm)].
- [ ] The `logging` library is being used and no `print` statements are left.

### CODE REVIEW

**Comment by szmazurek** [comment]
@sarthakpati check the output of the spellchecker action, this will probably need some more customization, I also have no idea why it complains about yaml extension

**Comment by sarthakpati** [comment]
Add a new file called `.spelling` to the repo with the following:

```yaml
words:
  - yml
  - yaml
  - gandlf
  - GaNDLF
```

Update the action to use this file:

```yaml
# previous steps go here
      - name: Spellcheck
        uses: check-spelling/check-spelling@main
        with:
          config: .spelling ## this is needed
```

**Comment by sarthakpati** [comment]
@szmazurek it seems like the CI is working, no? Can you please take a look and fix the misspelt words so that we can merge?

**Comment by sarthakpati** [comment]
Hey @szmazurek is this good to merge?

File: GANDLF/cli/huggingface_hub_handler.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def push_to_model_hub(
         ignore_patterns=ignore_patterns,
         delete_patterns=delete_patterns,
     )
-    print("Model Sucessfully Uploded")
+    print("Model Successfully Uploded")
```
Comment (sarthakpati) [inline GANDLF/cli/huggingface_hub_handler.py:124]:
```suggestion
    print("Model Successfully Uploaded")
```

Comment (szmazurek) [reply GANDLF/cli/huggingface_hub_handler.py]: added

**Review by sarthakpati — APPROVED** [review]

**Review by sarthakpati — APPROVED** [review]

**Pull request merged**